pkg/controllers/provisioning/nodepool_test.go

```diff
@@ -1459,4 +1470,361 @@ var _ = Describe("NodePool/Provisioning", func() {
 			})
 		})
 	})
+	Context("Resource Calculations", func() {
+
+		It("should schedule based on the sum of containers and sidecarContainers", func() {
+			pod := test.Pod(test.PodOptions{
+				ResourceRequirements: v1.ResourceRequirements{
+					Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("2"), v1.ResourceMemory: resource.MustParse("1Gi")},
+					Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("2")},
+				},
+
+				InitContainers: []v1.Container{
+					{
+						RestartPolicy: lo.ToPtr(v1.ContainerRestartPolicyAlways),
+						Resources: v1.ResourceRequirements{
+							Limits:   v1.ResourceList{v1.ResourceCPU: resource.MustParse("4"), v1.ResourceMemory: resource.MustParse("2Gi")},
+							Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1")},
+						},
+					},
+				},
+			})
+			podResources := resources.Ceiling(pod)
```

Comment:
I think we should make these scheduling tests as well. These seem like good unit tests to go under the `utils/resources/suite_test.go` package, but this package should be testing the ability of pods to schedule on instance types of certain sizes.

Here, for test test cases, I think we should be updating the instance types that are returned from the cloudprovider and ensuring that given the resource requests that expect from certain pods, that the pods still schedule to the instance types that we expect. Feel free to reach out to me if that doesn't make sense.

Still, I think these tests are really good and we should move them all into the `suite_test.go` in `pkg/utils/resources`